lang/cs/Org.Apache.REEF.Driver/Bridge/Events/FailedTask.cs

```diff
@@ -87,16 +89,29 @@ public Optional<IActiveContext> GetActiveContext()
             return Optional<IActiveContext>.Of(new ActiveContext(ActiveContextClr2Java));
         }
 
+        /// <summary>
+        /// Returns the Exception causing the Failed Task.
+        /// </summary>
+        /// <returns>the Exception causing the Failed Task.</returns>
+        /// <remarks>
+        /// If the Exception was caused by a control flow error (start, stop, suspend), 
+        /// a <see cref="TaskClientCodeException"/> is expected.
+        /// If the original Exception was not serializable, a <see cref="NonSerializableTaskException"/> is expected.
+        /// If the Exception was missing, presumably caused by a failed Evaluator, a 
+        /// <see cref="TaskExceptionMissingException"/> is expected.
+        /// </remarks>
         public Exception AsError()
         {
             return _cause;
         }
 
-        private static Exception GetCause(byte[] serializedCause, string taskExceptionString)
+        private static Exception GetCause(
+            byte[] serializedCause, string originalTaskExceptionToString)
```

Comment:
`serializedCause` might not be able to be deserialized. In other words, the serialization implementation was correct but the deserialization implementation was incorrect.